src/app/register/register-form.tsx

```diff
@@ -0,0 +1,121 @@
+"use client"
+
+import React, { useState } from "react"
+import Link from "next/link"
+import {
+  Session,
+  createClientComponentClient,
+} from "@supabase/auth-helpers-nextjs"
+import { Database } from "@/types/supabase"
+import { redirect } from "next/navigation"
+import Button from "@/components/common/Button"
+import { css } from "../../../styled-system/css"
+import Input from "@/components/common/Input"
+
+export default function AccountForm({ session }: { session: Session | null }) {
+  const supabase = createClientComponentClient<Database>()
+  const [formData, setFormData] = useState({
+    username: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  })
+  const { email, username, password, confirmPassword } = formData
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value })
+  }
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault()
+    if (password !== confirmPassword) {
+      alert("Passwords don't match")
+    } else {
+      const { error } = await supabase.auth.signUp({
```

Comment:
bug: On my side, when I login, the request succeded but I'm not redirected to any page. I believe it's missing the redirect after the response.